### ISSUE TITLE
display of last exit status displays too late.

### ISSUE THREAD

**Issue by markstos** [IlanCosman/tide#59]
#### Describe the bug

I expect the right hand exit status code to be reporting the command just run. However, it appears to lag by a command.

#### Steps to reproduce

1. Type `boom`

#### Expected behavior

A failure exit code is expected, but a geen checkmark is displayed instead.

You can now type `true` and press return, and a red x will appear. Ths is  `tide` is belatedly reporting failure from an earlier command.

This delay makes the exit status reporting confusing and not useful. If it can't be fixed to update the status of the previous command, it should be removed.

#### Environment (please complete the following information)

- Operating System: Arch Linux
- Output of 

```console
fish, version 3.1.2
tide, version 3.0.0
alacritty
```

**Comment by IlanCosman** [comment]
Obviously you're right. This is the strangest bug I've encountered in a long while, more to follow when I figure it out 😂I'll give you a sneak peak though, try out the Lean style, the status item works perfectly there. It seems to have something to do with the prompt_char item.

**Comment by markstos** [comment]
An unexpected trigger.  You'll have to add more config details to `tide bug-report` in there are more config-specific bugs.

**Comment by IlanCosman** [comment]
I still don't really understand why my changes fixed this. I'll look more into it. Thanks for submitting this bug report 😄

**Comment by IlanCosman** [comment]
Okay, I get it now. Mostly putting this here for me in the future, but you too if you're interested.

I was only refreshing the prompt when the `left_prompt_var` changed. This made perfect sense to me, as the `right_prompt_var` is always changed before the left, thus we can do one refresh instead of two. The (in retrospect) obvious problem is that sometimes I change the right prompt without changing the left, causing the prompt not to update when it should. There were a number of factors concealing this problem however.

First is the fact that when the prompt is two lines, updating the "right prompt" `status` is is actually updating the `left_prompt_var` because of the way that multi-line right prompt works. So when the prompt is two lines, the bug won't appear.

Secondly, the `prompt_char` item, which is always in the left prompt, and also responds to changes in the previous status by turning from green to red if the status was anything but 0. Thus, if the previous status was 0, and then we run a non-existent command, everything seems fine because the left `prompt_char` changed to red and the right prompt `status` changed to `127`. However, if our previous status wasn't 0, then the left `prompt_char` would already be red. Then, when we run a non-existent command the prompt won't refresh because the `--on-variable` handler only triggers when the variable is actually changed, and our `left_prompt_var` stayed the same.

So basically the only way to see the bug easily was a single-line prompt that doesn't have the `prompt_char` item, and I guess I just didn't spend enough time with those 😂
